main.tf

```diff
@@ -115,12 +115,41 @@ resource "aws_security_group_rule" "ingress_through_https" {
 #------------------------------------------------------------------------------
 # AWS LOAD BALANCER - Target Groups
 #------------------------------------------------------------------------------
+resource "null_resource" "lb_http_tgs_config" {
+  for_each = {
+    for name, config in var.http_ports : name => config
+    if lookup(config, "type", "") == "" || lookup(config, "type", "") == "forward"
+  }
+
+  triggers = {
+    # Store the md5 of the config so that if anything in `each.value` changes,
+    # the trigger changes and thus the resource changes.
+    config_md5 = md5(jsonencode(each.value))
+  }
+}
+
+resource "random_id" "lb_http_tgs_id" {
+  for_each = {
+    for name, config in var.http_ports : name => config
+    if lookup(config, "type", "") == "" || lookup(config, "type", "") == "forward"
+  }
+
+  byte_length = 2
+
+  lifecycle {
+    # Trigger a replacement whenever the configuration changes.
+    replace_triggered_by = [
+      null_resource.lb_http_tgs_config[each.key],
+    ]
+  }
+}
+
 resource "aws_lb_target_group" "lb_http_tgs" {
   for_each = {
     for name, config in var.http_ports : name => config
     if lookup(config, "type", "") == "" || lookup(config, "type", "") == "forward"
   }
-  name                          = "${var.name_prefix}-${each.key}-http-${each.value.target_group_port}"
+  name                          = "${var.name_prefix}-http-${each.value.target_group_port}-${random_id.lb_http_tgs_id[each.key].hex}"
```

Comment:
Not entirely sure yet. Removed `${each.key}-` to have more space for random id suffix but.

Reply:
I think it looks ok @nickreynke, the name will contain the port name. As long as it is easier to find and understand by users like you I think its fine!!!